Env 51 cards (#235)
sitevision/envision: packages/envision-docs/src/pages/components/cards.md

---
title: Cards
---

<span class="env-badge env-badge--info">Added in 9.0.1</span>

## Cards & Cardholders

> Use Cards for displaying information about for example a page, a group or a user.

Reviewer: ```suggestion
Use Cards to display information about, for example, a page, a group or a user.
```

A Card is a simple container that may have a header and a footer. A Card does not have

> it's own colors, so combine with `env-block` classes to set colors and `env-shadow` to set box shadow.

Reviewer: Tror man kan bryta upp denna mening i två. Något så här ish?

_A Card has no colors or shadows of its own. To add colors and shadows, you may instead combine the "env-block" and "env-shadow" classes._

                     src="https://unsplash.it/64/64/"
                     alt="Example profile image"
                  />
               </a>
            </div>
            <div class="env-media__body env-text--small">
               <p class="env-card__text">Open group &bull; 29 members</p>
               <p class="env-card__text">Latest activity 3 hours ago</p>
            </div>
         </article>

         <h2 class="env-card__text env-card__text--title">Lorem ipsum dolor</h2>
      </header>
      <div class="env-card__body">
         <p class="env-card__text env-card__text--secondary">
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent
            eget lobortis dui, in accumsan augue.
         </p>
      </div>
      <footer class="env-card__footer env-text--small">
         <a type="button" class="env-button env-button--link">
            <svg class="env-icon env-icon--small env-m-right--xx-small">
               <use
                  xlink:href="/sitevision/envision-icons.svg#icon-phone"
               ></use>
            </svg>
            Administrator
         </a>
      </footer>
   </article>
</div>
```

### Card anatomy

> A card may have a header, a body and a footer: `.env-card__header`, `.env-card__body`, `.env-card__footer`.

Reviewer: ```suggestion
A Card may have a header, a body and a footer: `.env-card__header`, `.env-card__body`, `.env-card__footer`.
```

> No matter source order, header will always display first and footer last. The footer

Reviewer: Vad menas med "source order"? Ordningen på HTML-markupen?

_header will always display first_ <-- känns om man kan skriva den delen tydligare, kommer inte på något bra exempel dock...

Author: Ja, ordning sätts av flexboxen så markup kan komma i vilken ordning som helst

> will always stick to the bottom of the card. Inside a Cardholder, if the card needs to stretch

Reviewer: ```suggestion
will always stick to the bottom of the Card. Inside a Cardholder, if the Card needs to stretch
```

> vertically, the body will be the element that stretches.

Reviewer: ```suggestion
vertically, the body of the Card will be the element that stretches.
```

```html
<div class="doc-demo-dark">
   <article class="example-card-parts env-card env-block env-shadow-small">
      <header class="env-card__header">Header</header>
      <div class="env-card__body">Body</div>
      <footer class="env-card__footer">Footer</footer>
   </article>
</div>
```

## Cardholder

A Cardholder will make a nice grid of Cards that adapt to the given space.

> Use one `env-cardholder__slot` to hold each card. See configuration options below.

Reviewer: ```suggestion
Use one `env-cardholder__slot` to hold each Card. See configuration options below.
```

            </div>
         </article>
      </div>
      <div class="env-cardholder__slot">
         <article class="env-card env-block env-shadow-small">
            <div class="env-card__body">
               <img
                  class="env-card__image env-profile-image env-profile-image--small"
                  src="https://unsplash.it/64/64/"
                  alt="Example profile image"
               />
               <h2 class="env-card__text env-card__text--title">
                  Lorem Ipsumsson
               </h2>
               <p class="env-card__text env-card__text--secondary">
                  Utvecklare
               </p>
               <p class="env-card__text env-card__text--secondary">
                  <a href="#" class="env-link env-link-secondary"
                     >lorem@domain.com</a
                  >
               </p>
               <p class="env-card__text env-card__text--secondary">
                  <a href="#" class="env-link env-link-secondary"
                     >012-345 67 89</a
                  >
               </p>
            </div>
         </article>
      </div>
   </div>
</div>
```

### Cardholder examples

> There are several modifiers for the cardholder that controls the cardholder slots:

Reviewer: ```suggestion
There are several modifiers for the Cardholder that controls the Cardholder slots:
```

> `.env-cardholder--N` - where N is a number that controls how many slots
> the cardholder should display at most in one row.

Reviewer: ```suggestion
`.env-cardholder--N` - where N is a number that controls how many slots
the Cardholder can display at most in one row.
```

> Possible values for N is 2, 3, 4, 5, 6, 8.

Reviewer: Borde värdena "poppas" ut med code-formatering ex. `2`, `3`, `4`

> `.env-cardholder--max` - By default, slots have no max-width. This modifier sets a max width.

Reviewer: `max-width` eller `max width`?

Author: Båda är ju rätt, men kanske bättre att skriva icke-kod i meningen?

Reviewer: Ja eller kanske märka ut med formatering när det handlar om kod?

Slots will be centered in the last row and may not align to the columns. On small screens, the slots
may not fill the entire screen width.

> `.env-cardholder--start` - Must be combined with `.env-cardholder--max`. Will align slots left.

Reviewer: ```suggestion
`.env-cardholder--start` - Must be combined with `.env-cardholder--max`. Will align slots to the left.
```

To control the size of the columns there are two CSS variables that may be set:

```CSS
.your-cardholder-container {
   /*

>    min-width is always used. Will control when breaks occur.

Reviewer: Kan man på något vis förklara vilka "breaks" det handlar om? För att förtydliga.

   max-width is only used when modifier env-cardholder--max is set.
   */
   --env-cardholder-column-min-width: 240px;
   --env-cardholder-column-max-width: 400px;
}
```

> Below are a few examples that you may resize and see how the columns adapt to the given space.

Reviewer: Det här var så snyggt!!!! Love this! 💯 😍 🔥
